k8s/core/services.go

```diff
@@ -63,6 +67,31 @@ func (c *Client) GetService(svcName string, svcNS string) (*corev1.Service, erro
 	return c.core.Services(svcNS).Get(svcName, metav1.GetOptions{})
 }
 
+// ListServices list services using filters or list all if options are empty
+func (c *Client) ListServices(svcNamespace string, listOptions metav1.ListOptions) (*corev1.ServiceList, error) {
+	if err := c.initClient(); err != nil {
+		return nil, err
+	}
+
+	return c.core.Services(svcNamespace).List(listOptions)
+}
+
+// GetExternalIP gets the externalIP of a loadBalancer present in a given namespace, returns empty if LoadBalancer not
+// found
+func (c *Client) GetExternalIP(namespace string) (string, error) {
```

Comment:
Instead of just getting the external ip like this we should get service's endpoint. 
Basically given a service name and namespace, we should return cluster ip if the service type is ClusterIP (and if cluster ip is not none), ingress ip if the type is LoadBalancer. Let's leave node port out for now.